test/libyul/yulSyntaxTests/invalid/invalid_debug_merge.yul

```diff
@@ -0,0 +1,7 @@
+object "object" {
+    code {
+        /// @debug.merge {"HELLO": 2
+    }
+}
+// ----
+// SyntaxError 5721: (37-65): @debug.merge: Could not parse debug data: parse error at line 1, column 12: syntax error while parsing object - unexpected end of input; expected '}'
```

Comment:
Will this *always* error out now on invalid ``@debug.*`` attributes? What happened before if we had this kind of attribute? If this compiled before the change, we need to wonder if this is breaking (and should only be parsed if enabled somehow) - even though it may be fine in practice. But can you explain how this behaved before the PR?

Reply:
@cameel what would you say to this? this is technically breaking, but would you say its ok?

Reply:
I think I will just base on https://github.com/ethereum/solidity/pull/15289. So attribute parsing is only enabled, if ethdebug was requested.

Reply:
Looks like this would be ignored before and now it is an error. I checked now and when we introduced the `@src` tag we did something like this too - solc 0.8.8 started issuing errors on invalid `@src` tags. So there is a precedent for ignoring such minor breaking changes.

Similarly in #15209, we were careful not to be completely breaking, but it was still too constraining to keep it 100% non-breaking for every minor irrelevant detail - e.g. now single quotes are supported and you'll get an error on an unmatched quote, which previously would have been not recognized as the beginning of a string and ignored.

So I'd say it's fine if the alternative requires too much effort. It should still be documented in the changelog though. Looks like you found a good alternative though - parsing it only with ethdebug enabled - so I'd go with that.